core/base/contourTreeAlignment/ContourTreeAlignment.h

```diff
@@ -298,7 +298,7 @@ int ttk::ContourTreeAlignment::execute(const vector<void *> &scalarsVP,
   }
   std::srand(seed);
   if(alignmenttreeType != lastMatchedValue)
-    std::random_shuffle(permutation.begin(), permutation.end());
+    std::shuffle(permutation.begin(), permutation.end(), std::random_device{});
```

Comment:
Does this way of shuffling also use the fixed seed from above? (Is it influenced by std::srand(seed)?)

Reply:
Good call @floWetzels! Indeed, std::shuffle uses a more robust random number generator than std::random_shuffle, that's why I decided to replace it.
However, the new random generator doesn't use the seed set by std::srand. I tried several random engines but it appears to have consequences on the `contourTreeAlignment.pvsm` state file in the `ttk-data` repository.
For now, I think the best way is to revert this change and keep the old (but seeded) random generator.
Thanks for pointing that and sorry for the inconvenience.

Reply:
No problem, maybe we find a better solution soon.
The other changes in the ContourTreeAlignment class seem reasonable to me and should not change anything. So that's fine then.